CONTRIBUTING.md

```diff
@@ -1,41 +1,125 @@
+    * <a href="#local-copy">Make a local copy of Gearpump</a>
+    * <a href="#build">How to build</a>
+    * <a href="#test">How to test</a>
+    * <a href="#build-doc">How to build document</a>
+    * <a href="#IDE-setup">IDE setup</a>
+    * <a href="#code-style">Code style</a>
+    * <a href="#write-unittest">How to write unit test</a>
+    * <a href="#write-integrationtest">How to write integration test</a>
+    * <a href="#write-doc">How to write document</a>
+* <a href="#committer-work">Committer section</a>
+    * <a href="#approve-pull-request">Approve a pull request</a>
+    * <a href="#merge-pull-request">Merge a pull request or patch</a>
+    * <a href="#release">How to make a release</a>    
 
-3. In local master branch, periodically sync the forked master with the main master with 
- 
-  ```
-   git pull --rebase upstream  master
-   git push origin master
-  ``` 
-No work should ever be done in the forked master. Another way to do this is to 
+---
 
- ```
- git checkout master
- git fetch upstream
- git rebase upstream/master
- ```
+<a name="welcome"></a>
+
+# Welcome!
+
+If you are reading this document then you are interested in contributing to the Gearpump project -- many thanks for that!
+All contributions are welcome: ideas, documentation, code, patches, bug reports, feature requests, etc. 
+
+
+<a name="workflow"></a>
+# Contribution workflow
+
+This section explains how to make a contribution. 
+
+
+<a name="report-bug"></a>
+
+## Report a bug
+
+To report a bug you should [open an issue](https://issues.apache.org/jira/browse/GEARPUMP) in our issue tracker that
+summarizes the bug.  Set the form field "Issue type" to "Bug".  If you have not used the issue tracker before you will
+need to register an account (free), log in, and then click on the blue "Create Issue" button in the top navigation bar.
+
+In order to help us understand and fix the bug it would be great if you could provide us with:
+
+1. The steps to reproduce the bug.  This includes information about e.g. the Gearpump version you were using, the deployment model, etc.
+2. The expected behavior.
+3. The actual, incorrect behavior.
+
+Feel free to search the issue tracker for existing issues (aka tickets) that already describe the problem;  if there is
+such a ticket please add your information as a comment.
+
+**If you want to provide a patch along with your bug report:**
+That is great!  In this case please send us a pull request as described in section [Create a pull request](#create-pr) below.
+You can also opt to attach a patch file to the issue ticket, but we prefer pull requests because they are easier to work
+with.
+
+
+<a name="request-feature"></a>
+
+## Request a new feature
 
-4. Create a working branch
+To request a new feature you should [open an issue](https://issues.apache.org/jira/browse/GEARPUMP) in our issue tracker
+and summarize the desired functionality.  Set the form field "Issue type" to "New feature".  If you have not used the
```

Comment:
these sections are borrowed from Storm guide. I am not sure whether we want to have all these assignable by contributor or we shall adjust it later by committers?

And I prefer to let committer adjust the "Priority" field instead of committer.